components/resolved_ts/src/resolver.rs

```diff
@@ -1,13 +1,40 @@
 // Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
 
-use std::{cmp, collections::BTreeMap, sync::Arc};
+use std::{
+    cmp,
+    cmp::Reverse,
+    collections::{BTreeMap, BinaryHeap},
+    sync::Arc,
+};
 
 use collections::{HashMap, HashSet};
 use raftstore::store::RegionReadProgress;
 use txn_types::TimeStamp;
 
 use crate::metrics::RTS_RESOLVED_FAIL_ADVANCE_VEC;
 
+#[derive(Debug, Clone, Copy)]
+pub struct ResolvedTs {
+    pub raw_ts: TimeStamp,
+    pub txn_ts: TimeStamp,
+}
+
+impl ResolvedTs {
+    pub fn default() -> ResolvedTs {
+        ResolvedTs {
+            raw_ts: TimeStamp::zero(),
+            txn_ts: TimeStamp::zero(),
+        }
+    }
+    pub fn get_min(&self) -> TimeStamp {
+        cmp::min(self.raw_ts, self.txn_ts)
```

Comment:
Does it need to skip `TimeStamp::zero()`?

Reply:
`raw_ts` and `txn_ts` are initialized as `TimeStamp::zero()` and updated every MinTs task, they all never decrease. So I think they must not be `zero` returned from 'Resolver.resolve'.